pkg/kubelet/lifecycle/predicate.go

```diff
@@ -64,6 +74,18 @@ func (w *predicateAdmitHandler) Admit(attrs *PodAdmitAttributes) PodAdmitResult
 		}
 	}
 	if !fit {
+		fit, reasons, err = w.admissionFailureHandler.HandleAdmissionFailure(pod, reasons)
+		if err != nil {
+			message := fmt.Sprintf("HandleAdmissionFailure failed due to %v, which is unexpected.", err)
```

Comment:
I changed the message to: fmt.Sprintf("Unexpected error while attempting recover from admission failure: %v", err)
I think that is sufficient, as more specific information should be returned by the error from HandleAdmissionFailure.  For example, "Failed to preempt pods %v to make room for pod %v with error: failed to evict pod" would provide sufficient information for someone debugging to know exactly what happened.